src/NHibernate/Transform/QueryAliasToObjectPropertySetter.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace NHibernate.Transform
+{
+	[Serializable]
+	public class QueryAliasToObjectPropertySetter
+	{
+		private readonly IEnumerable<FieldInfo> _fields;
+		private readonly IEnumerable<PropertyInfo> _properties;
+
+		private QueryAliasToObjectPropertySetter(FieldInfo[] fields, PropertyInfo[] properties)
+		{
+			_fields = fields;
+			_properties = properties;
+		}
+
+		public static QueryAliasToObjectPropertySetter MakeFor(System.Type objType)
+		{
+			var bindingFlags = BindingFlags.Instance |
+				BindingFlags.Public |
+				BindingFlags.NonPublic |
+				BindingFlags.IgnoreCase;
+			var fields = objType.GetFields(bindingFlags);
+			var properties = objType.GetProperties(bindingFlags);
```

Comment:
When getting fields or properties without filtering by a name, what is the point of `BindingFlags.IgnoreCase`?